### PR TITLE
add compileall stubs to 3 and add types to the ones in 2

### PR DESCRIPTION
**Pull request by JelleZijlstra**
Didn't merge the stubs because all functions have additional parameters since 3.2,
so there would be no shared code between 2 and 3.

### CODE REVIEW

**Comment by ambv** [comment]
Good call!

File: stdlib/3/compileall.pyi [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+if sys.version_info < (3, 6):
+    _Path = Union[str, bytes]
+else:
+    _Path = Union[str, bytes, os.PathLike]
+
+# fx can be any object with a 'search' method; once we have Protocols we can change the type
+if sys.version_info < (3, 5):
+    def compile_dir(dir: _Path, maxlevels: int = ..., ddir: _Path = ..., force: bool = ..., rx: Optional[Pattern] = ..., quiet: int = ..., legacy: bool = ..., optimize: int = ...) -> None: ...
+else:
+    def compile_dir(dir: _Path, maxlevels: int = ..., ddir: _Path = ..., force: bool = ..., rx: Optional[Pattern] = ..., quiet: int = ..., legacy: bool = ..., optimize: int = ..., workers: int = ...) -> None: ...
```
Comment (JukkaL) [inline stdlib/3/compileall.pyi]: According to https://docs.python.org/3/library/compileall.html, `ddir` is `Optional`.

File: stdlib/3/compileall.pyi [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+def compile_file(fullname: _Path, ddir: _Path = ..., force: bool = ..., rx: Optional[Pattern] = ..., quiet: int = ..., legacy: bool = ..., optimize: int = ...) -> None: ...
```
Comment (JukkaL) [inline stdlib/3/compileall.pyi]: Again, `ddir` is optional (probably everywhere in this module).

File: stdlib/2/compileall.pyi [hunk 3]
```diff
@@ -1,7 +1,10 @@
...
 
-def compile_dir(dir, maxlevels=..., ddir=..., force=..., rx=..., quiet=...): ...
-def compile_file(fullname, ddir=..., force=..., rx=..., quiet=...): ...
-def compile_path(skip_curdir=..., maxlevels=..., force=..., quiet=...): ...
+from typing import Optional, Pattern, Union
+
+_Path = Union[str, bytes]
+
+# fx can be any object with a 'search' method; once we have Protocols we can change the type
+def compile_dir(dir: _Path, maxlevels: int = ..., ddir: _Path = ..., force: bool = ..., rx: Optional[Pattern] = ..., quiet: int = ...) -> None: ...
```
Comment (JukkaL) [inline stdlib/2/compileall.pyi]: Looks like `ddir` should be `Optional`.

**Comment by JelleZijlstra** [comment]
Opening a new PR to address Jukka's comments.

**Pull request merged**